authzed/api/v1/experimental_service.proto

```diff
@@ -44,19 +44,67 @@ service ExperimentalService {
       };
     }
 
+  // NOTE: BulkCheckPermission has been promoted to the stable API as "CheckBulkPermission" and the
+  // API will be removed from experimental in a future release.
   rpc BulkCheckPermission(BulkCheckPermissionRequest)
     returns (BulkCheckPermissionResponse) {
       option (google.api.http) = {
         post: "/v1/experimental/permissions/bulkcheckpermission"
         body: "*"
       };
+      option deprecated = true;
+    }
+
+  // EXPERIMENTAL: ReflectSchema is an API that allows clients to reflect the schema stored in
+  // SpiceDB. This is useful for clients that need to introspect the schema of a SpiceDB instance.
+  rpc ExperimentalReflectSchema(ExperimentalReflectSchemaRequest)
+    returns (ExperimentalReflectSchemaResponse) {
+      option (google.api.http) = {
+        post: "/v1/experimental/reflectschema"
+        body: "*"
+      };
+    }
+
+  // EXPERIMENTAL: ComputablePermissions is an API that allows clients to request the set of
+  // permissions that compute based off a set of relations. For example, if a schema has a relation
+  // `viewer` and a permission `view` defined as `permission view = viewer + editor`, then the
+  // computable permissions for the relation `viewer` will include `view`.
+  rpc ExperimentalComputablePermissions(ExperimentalComputablePermissionsRequest)
```

Comment:
this name confused me on the first read (and the inverse one) - can you explain why this API is better than e.g. returning an expression tree?

Reply:
Because understanding how to walk the expression tree is a non-trivial task, especially as we add more operations to the schema

Reply:
It would just be a AST and the leaf nodes will all be relations, right? So even as we add operations we still just have a tree with relations as leaves. Internal nodes are `(op, []children)` where a child is either another internal node or a leaf (relation) node?

Reply:
I still think we could come up with an AST that:

- has enough info that consumers can understand the shape of the expressions
- is simple enough that you can use it to determine the answers to basic questions (like what permission is affected by what relation) without needing to understand the full details of the tree

but this is experimental and nothing stops us from adding that in the future, so I'll leave it.